### PR TITLE
CB-17474: Added flow to beginning of DL resize which validates if Atlas is up to date.

### PR DESCRIPTION
**Pull request by sxxgrc**
Jira: https://jira.cloudera.com/browse/CB-17474

The objective is to use the salt changes from CB-17112 to have an automatic flow step at the beginning of resize, which will verify if Atlas is up to date before continuing with the resize.
In the case that Atlas is not up to date, the flow will wait for a specified time until it is up to date or the process times out.

This change consists of:

1. Flow changes within the Datalake code which is accompanied by the expected addition to the resize flow chain
2. Flow changes in the Cloudbreak core code, which is exposed via a stackv4endpoint method
3. New logic in cloudbreak core which generates the salt configuration needed for the Atlas salt changes to run

Please note that I am new to these changes and am making a lot of assumptions, specifically basing all of my changes off the backup/restore changes. So if there is anything incorrect or something that could be better, please let me know and I will make the modification.

### CODE REVIEW

**Comment by sxxgrc** [comment]
Local testing has been successful so I am moving this to a real PR.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/doc/OperationDescriptions.java [hunk 1]
```diff
@@ -89,6 +89,7 @@ public static class StackOpDescription {
         public static final String RANGER_RAZ_ENABLED = "Determines if Ranger Raz is present in the cluster.";
         public static final String GENERATE_IMAGE_CATALOG = "Generates an image catalog that only contains the currently used image for creating instances";
         public static final String RE_REGISTER_CLUSTER_PROXY_CONFIG = "Re-registers the cluster proxy config for the cluster if needed.";
+        public static final String CHECK_ATLAS_UPDATED = "Checks if the Atlas service on the cluster is up to date with recent lineage changes.";
```
Comment (kkalvagadda1) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/doc/OperationDescriptions.java:92]: How about "Checks if the Atlas service on the cluster is up to date with the lineage.";

**Comment by horadla23** [comment]
please use draft PR instead of "waiting-for-test" label if it is not ready to merge

**Comment by sxxgrc** [comment]
I apologize @horadla23 that label should have been removed over a week ago. This is ready to be reviewed although an ongoing conversation about this is occurring within the Jira comments.
Feel free to look at the comments to better understand the discussion before making a review.

**Review by RokolyA94 — APPROVED** [review]
LGTM

**Review by lnardai — CHANGES_REQUESTED** [review]
See discussion on CB ticket

**Review by doktoric — CHANGES_REQUESTED** [review]
[CB-17474](https://jira.cloudera.com/browse/CB-17474) where this discussion happening

**Review by akanto — CHANGES_REQUESTED** [review]
I don't understand this pull request, I think we have clearly explained in the jira that we will not accept this and this should be part of the CM, since Atlas management is not the task of Cloudbreak and we deal with infra only.

https://jira.cloudera.com/browse/CB-17474?focusedCommentId=5343564&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-5343564

**Pull request closed without merging**